### ISSUE TITLE
Add liquid support for *.md files

### ISSUE THREAD

**Issue by Yuudaari** [panoply/vscode-liquid#10]
Jekyll/GitHub Pages support `.markdown` and `.md` files. It would be sweet if this extension could support them.

**Comment by panoply** [comment]
Interesting, I've never seen liquid infused markdown files, can you give me some use cases (just for reference sake?) – Should be easy to implement, will put it in the road map.

**Comment by Yuudaari** [comment]
For me, writing a blog post/chapter/etc is easiest in markdown format, whereas writing the rest of the site is easier in pure HTML. So whether or not this extension supports markdown, I'll still be using it. 

Due to it not being supported (only the frontmatter is highlighted, I'm not sure where that comes from, whether it's this extension or sth else), I've built my site intentionally to have no liquid in the markdown. Instead, rather, the structure of my site is as follows:

1. A "chapter" layout, which is liquid html (and has a lot of the really complex functionality in it)
2. For posts/chapters, I just write in markdown and set the layout to "chapter" (which sticks the markdown content inside all the html)

I had a workaround in this case, to use the layout HTML & frontmatter variables for all the liquid I needed, so I don't really have anything to show for use cases (yet). It just so happened that I noticed the snippets weren't supported, then I realised highlighting wasn't either, and then I figured it was an issue worth reporting.

So this isn't a priority at all, for me anyway. If I think of any, or run into a use case, I'll add it to this issue if you like.

PS: Is this kind of thing fixable? Is it worth making an issue for? 
![image](https://user-images.githubusercontent.com/6081834/57298076-ed4a8500-7096-11e9-9950-784a2be1713c.png)

PPS: Jekyll also supports `scss`/`sass` files, in which the frontmatter actually throws an error right now. Of course, liquid is a little pointless in sass code, so maybe that'd be out of the range of this extension?

**Comment by panoply** [comment]
> only the frontmatter is highlighted, I'm not sure where that comes from.

This extension supports YAML markdown between `---` tags

I think I understand what you mean, you would like the markdown syntax to be highlighted,  opposed to liquid syntax within markdown files? –  If this is the case, you could can just install markdown syntax support extension from marketplace, it should suffice but if it give you errors due to the triple dash wrappers (`---`) then I can look into potentially look into supporting markdown. 

Regarding your issue with the link relative syntax, try single quotes or no quotes opposed to double, eg:

```html
<link rel="stylesheet" href="{{ '/index.css' | relative_url }}" />
<!-- OR -->
<link rel="stylesheet" href="{{ /index.css | relative_url }}" />
```

Regarding SCSS / SASS.
Liquid syntax support within these files is in the roadmap as Shopify theme developers tend to infuse their sass code with liquid tags, but with that said I'm not keen on encouraging this type of thing.

**Comment by Yuudaari** [comment]
> I think I understand what you mean, you would like the markdown syntax to be highlighted, opposed to liquid syntax within markdown files? – If this is the case, you could can just install markdown syntax support extension from marketplace, it should suffice but if it give you errors due to the triple dash wrappers (`---`) then I can look into potentially look into supporting markdown.

Ah, no, you were correct before, sorry, I didn't mean to make this more confusing. The markdown is highlighted, and the frontmatter is highlighted, and there's no errors. However, this extension's snippets don't work (since they're only in for HTML, it looks like), the liquid doesn't highlight, etc etc.

> Regarding your issue with the link relative syntax

w/ this, I just meant to say that the liquid syntax highlighting doesn't work within html attribute values. That was what I was asking if it was possible to fix. However, I know the tmlanguage stuff has a lot of quirks, so I wasn't sure if it was possible. (Last time I touched the tmlanguage stuff in VSCode was a couple years ago now)

**Comment by panoply** [comment]
> Ah, no, you were correct before, sorry, I didn't mean to make this more confusing. The markdown is highlighted, and the frontmatter is highlighted, and there's no errors. However, this extension's snippets don't work (since they're only in for HTML, it looks like), the liquid doesn't highlight, etc etc.

Does File Association (within settings.json) not suffice in this case? – or do you mean intellisense?

> Liquid syntax highlighting doesn't work within html attribute values. 

I have considered looking into this. It's definitely possible and worth attempting.

> Last time I touched the tmlanguage stuff in VSCode was a couple years ago now

tmLang is still a cold heartless bitch.

**Comment by Yuudaari** [comment]
For file associations, what are you suggesting I associate it with? If I associate it with anything but MD I lose the MD syntax highlighting, I lose the ability to open a markdown preview, etc

**Comment by panoply** [comment]
Regarding attribute syntax highlighting, I just published support for this in version 1.6.4, if you update the extension you will get syntax highlighting within attribute strings, eg:

<img width="962" alt="Screenshot 2019-05-07 at 23 53 08" src="https://user-images.githubusercontent.com/7041324/57335567-4cee7200-7123-11e9-9351-afa7d2967a3b.png">

**Comment by panoply** [comment]
> For file associations, what are you suggesting I associate it with? If I associate it with anything but MD I lose the MD syntax highlighting, I lose the ability to open a markdown preview, etc

Ignore this, I didn't read your question properly.

**Comment by panoply** [comment]
@Yuudaari could you try explain once more what it is exactly you need? Screenshots would help, still having trouble understanding what you mean.

**Comment by Yuudaari** [comment]
By default, only the frontmatter is highlighted, and the same snippets don't work:
![2019-05-09_20-26-51](https://user-images.githubusercontent.com/6081834/57496525-2a7e6500-7299-11e9-8f8b-f84a0bbd1ddd.gif)

If I change the language to HTML (same as file association):
![2019-05-09_20-33-44](https://user-images.githubusercontent.com/6081834/57496681-ed66a280-7299-11e9-8c1e-4f8588bb50dd.gif)

I don't know of any way I could make it associate to both markdown and liquid simultaneously, besides thru an extension like this one injecting into the language.

As for the `scss` question, I just want this first error to go away:
![image](https://user-images.githubusercontent.com/6081834/57496895-dffde800-729a-11e9-9a70-4c15d6e78c4a.png)

After thinking about it more, I'm thinking it'd be impossible tho. I'm assuming that you'd have to override scss extension with your own impl, maybe somehow include the scss extension within your handling of the file? Or something weird and hacky like that.

**Comment by panoply** [comment]
Ah! I see. It's not impossible, just tedious. I am somewhat exploring the same hurdle bringing support to `.scss` files, as shopify theme developers infuse liquid tags as sass variables. Currently, I can achieve this  by wrapping the specified contents within block comment tags, for example:

```
<!-- liquid -->

<!-- endliquid -->
```

From here, I can inject the syntax, which will highlight all code between these comment blocks as you would expect, the downside is you will have comment wrapping blocks.

**Comment by panoply** [comment]
Solved in v1.8.0 – The extension now support liquid infused within `.md` files.
